### DISCUSSION TITLE
How to load scripts only on a specific page?

### DISCUSSION THREAD

**Discussion by viperfx** [vercel/next.js#32335] [Help]
I have used both import `Script from "next/script";` and `import Head from "next/head";` method to load two scripts on pages/index.js. Both of these options produce the same result.

The scripts seem to stay loaded after I navigate to other pages as well. How can I make sure the scripts only load only on the index.js and no others?

**Comment by webdeb** [comment]
@viperfx have you found a satisfying solution?

**Comment by fescobar** [comment]
Any answer about this?

**Comment by DutchPrince** [comment]
You can just load the next/script <Script /> tag on any page, which will load this script only on this page.

**Reply by aksjdjkasdjk** [reply]
But when we back from that page the script tag is applied for all the pages

**Comment by stephenasuncionDEV** [comment]
Any official solution?

**Comment by zishanaxioned** [comment]
This is what I followed for now. As there are no official doc mentioning this and found other documents irrelevant/not-working

`

import { useEffect } from 'react';
import { useRouter } from 'next/router';

const ExternalScript = () => {
  const router = useRouter();

  useEffect(() => {
    const pagePath = router.pathname === '/path';

    if (pagePath) {
      const script = document.createElement('script');
      script.id = 'id';
      script.innerHTML = `console.log('Script loaded');`;
      document.body.appendChild(script);

      return () => {
        document.getElementById('id')?.remove();
      };
    } else {
      document.getElementById('demand-base')?.remove();
    }
  }, [router.pathname]);

  return null;
};

export default ExternalScript;

`

**Reply by zishanaxioned** [reply]
Not sure why the code is not being indented properly. but you can take the above reference and give it a try
